src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/create/action/ResourcesAllocationAction.java

```diff
@@ -310,26 +312,16 @@ private void allocateProtectedPath(FlowCreateFsm stateMachine) throws Unroutable
         });
     }
 
-    private void updateIslsForFlowPath(FlowPath flowPath) throws ResourceAllocationException {
-        for (PathSegment pathSegment : flowPath.getSegments()) {
-            log.debug("Updating ISL for the path segment: {}", pathSegment);
-
-            updateAvailableBandwidth(pathSegment.getSrcSwitchId(), pathSegment.getSrcPort(),
-                    pathSegment.getDestSwitchId(), pathSegment.getDestPort());
-        }
-    }
-
-    private void updateAvailableBandwidth(SwitchId srcSwitch, int srcPort, SwitchId dstSwitch, int dstPort)
-            throws ResourceAllocationException {
-        long usedBandwidth = flowPathRepository.getUsedBandwidthBetweenEndpoints(srcSwitch, srcPort,
-                dstSwitch, dstPort);
-        log.debug("Updating ISL {}_{}-{}_{} with used bandwidth {}", srcSwitch, srcPort, dstSwitch, dstPort,
-                usedBandwidth);
-        long islAvailableBandwidth =
-                islRepository.updateAvailableBandwidth(srcSwitch, srcPort, dstSwitch, dstPort, usedBandwidth);
-        if (islAvailableBandwidth < 0) {
-            throw new ResourceAllocationException(format("ISL %s_%d-%s_%d was overprovisioned",
-                    srcSwitch, srcPort, dstSwitch, dstPort));
+    private void updateIslsForFlowPath(PathId pathId) throws ResourceAllocationException {
+        Map<IslEndpoints, Long> updatedIsls = islRepository.updateAvailableBandwidthOnIslsOccupiedByPath(pathId);
+        for (Entry<IslEndpoints, Long> entry : updatedIsls.entrySet()) {
+            IslEndpoints isl = entry.getKey();
+            log.debug("ISL {}_{}-{}_{} was updated with used bandwidth",
```

Comment:
same

Reply:
Right, the correct log message is in IslRepository. This excessive one has been removed.